core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -1,44 +1,100 @@
 package org.testcontainers.containers.wait;
 
+import lombok.extern.slf4j.Slf4j;
 import org.rnorth.ducttape.TimeoutException;
 import org.rnorth.ducttape.unreliables.Unreliables;
 import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.Container;
 import org.testcontainers.containers.ContainerLaunchException;
 import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.dockerclient.ProxiedUnixSocketClientProviderStrategy;
 
-import java.io.IOException;
 import java.net.Socket;
+import java.util.List;
+import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 
 /**
  * Waits until a socket connection can be established on a port exposed or mapped by the container.
  *
  * @author richardnorth
  */
+@Slf4j
 public class HostPortWaitStrategy extends GenericContainer.AbstractWaitStrategy {
+
+    private static final String SUCCESS_MARKER = "TESTCONTAINERS_SUCCESS";
     @Override
     protected void waitUntilReady() {
         final Integer port = getLivenessCheckPort();
         if (null == port) {
+            log.debug("Liveness check port of {} is empty. Not waiting.", container.getContainerName());
             return;
         }
 
-        final String ipAddress = container.getContainerIpAddress();
+        Callable<Boolean> checkStrategy;
+
+        // Special case for Docker for Mac, see #160
+        if (DockerClientFactory.instance().isUsing(ProxiedUnixSocketClientProviderStrategy.class)) {
+            List<Integer> exposedPorts = container.getExposedPorts();
+
+            Integer exposedPort = exposedPorts.stream()
```

Comment:
unfortunately, we have to do some reverse engineering to find the port originally exposed since it's not private (protected, in fact)